src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -106,7 +106,7 @@ private BoundExpression CompareWithConstant(BoundExpression input, BoundExpressi
                 );
         }
 
-        BoundExpression MakeDeclarationPattern(SyntaxNode syntax, BoundExpression input, LocalSymbol target, bool requiresNullTest)
+        BoundExpression MakeDeclarationPattern(SyntaxNode syntax, BoundExpression input, BoundExpression target, bool requiresNullTest)
```

Comment:
If this method is still expecting to only take locals as targets, consider adding an assertion to that fact.